packages/daimo-api/src/router.ts

```diff
@@ -41,9 +37,32 @@ export function createRouter(
   paymaster: Paymaster,
   faucet: Faucet,
   notifier: PushNotifier,
-  accountFactory: AccountFactory
+  accountFactory: AccountFactory,
+  telemetry: Telemetry
 ) {
-  return router({
+  // TODO: tracer doesn't work.
+  // https://github.com/honeycombio/honeycomb-opentelemetry-node/issues/239
+  // const tracer = trace.getTracer("daimo-api");
+
+  const tracerMiddleware = trpcT.middleware(async (opts) => {
+    // const span = tracer.startSpan(`trpc.${opts.type}`);
+    // span.setAttributes({ "trpc.path": opts.path });
+    const startMs = performance.now();
+
+    const result = await opts.next();
+
+    const endMs = performance.now();
+    telemetry.recordRpc(opts.ctx, opts.path, result.ok, endMs - startMs);
```

Comment:
out of curiosity since I don't see `async` keywords here, anything in this record doesn't block responses to the API right?

Reply:
nope just queues it up to be submitted